src/Chapter2.hs

```diff
@@ -348,9 +365,10 @@ Implement a function that returns only the first half of a given list.
 >>> firstHalf "bca"
 "b"
 -}
--- PUT THE FUNCTION TYPE IN HERE
-firstHalf l = error "firstHalf: Not implemented!"
-
+firstHalf :: [a] -> [a]
+firstHalf l =
+  let half = (length l) `div` 2
+  in subList 0 (half - 1) l
```

Comment:
Great usage of `let` construction! Makes code easily readable 💯 
I see that you reuse the `sublist` function in here, it is clever 👍🏼 
But also note, that `sublist 0` is actually the standard `take` function 🙂